helix-term/src/ui/popup.rs

```diff
@@ -115,8 +137,10 @@ impl<T: Component> Popup<T> {
             rel_x = rel_x.saturating_sub((rel_x + width).saturating_sub(viewport.width));
         }
 
-        let can_put_below = viewport.height > rel_y + height;
-        let can_put_above = rel_y.checked_sub(height).is_some();
+        // TODO can_put_below should be useless since MIN_HEIGHT is 4 and there is always 4 lines
+        // below the cursor (ask a maintainer and remove the code related to can_put_below).
+        let can_put_below = viewport.height > rel_y + MIN_HEIGHT;
```

Comment:
Why are there always 4 lines below the cursor? The statusline and promp would be just two (also add that to this comment, if it's not clear to me it's not clear to other readers)

Reply:
Right now I am assuming that the popup has a minimum size of 4. (we can make it configurable). only if this minimum Size is not met, we change the location of the popup. This actually what this PR about. IMO this improves the popup location much. (Right now, there is some cases where the popup changes the location without important reason)

In that comment, I said, since the minimum size of the popup can be 4 and below the cursor I have always at least 6 free lines, where the popup can be drawn (see screenshot below). So I think that the flag can_put_below is useless. 
![popup](https://github.com/helix-editor/helix/assets/15311024/5e8e17db-f947-421d-8467-7390cb1e8a18)

nevertheless I think we should keep that flag. maybe in the future we can make the minimum popup size configurable and some popups will request higher minimum size. Than this flag will be important.
(I remove that comment for the time being)

Reply:
hmm yeah almost everything your are saying is what I got from the code too just the following part:

> below the cursor I have always at least 6 free lines, where the popup can be drawn (see screenshot below)

this is not true. I think you are talking about scrolloff here but that can be configured/disabled (`:set scrolloff 0` so that assumption does not hold in general. 

So I would just remove that comment.